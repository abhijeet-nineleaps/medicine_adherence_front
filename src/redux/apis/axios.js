import defaultAxios from 'axios';
import { API_URL } from '../../repositories/var';

const axios = defaultAxios.create({
  baseURL: `${API_URL}`,
  headers: {'Content-Type': 'application/json'},
});

export default axios;
