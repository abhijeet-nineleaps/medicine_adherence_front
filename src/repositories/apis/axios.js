import defaultAxios from 'axios';
import { API_URL } from '../var';

const axios = defaultAxios.create({
  baseURL: `${API_URL}`,
  headers: {'Content-Type': 'application/json'},
});

export default axios;
