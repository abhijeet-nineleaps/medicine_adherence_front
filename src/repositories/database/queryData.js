const queryData = {
  getusermeds: async (txn) => {
    const reminder_array = [];
    return new Promise(resolve => {
      txn.executeSql(
        'SELECT * FROM `User_medicines`',
        [],
        function (_tx, res) {
          for (let i = 0; i < res.rows.length; ++i) {
            reminder_array.push(res.rows.item(i));
          }
          resolve(reminder_array);
        },
      );
    });
  },
};

export default queryData;
