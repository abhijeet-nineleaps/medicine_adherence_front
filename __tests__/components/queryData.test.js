import React from "react";
import enableHooks from "jest-react-hooks-shallow";
import Enzyme, { shallow } from 'enzyme';
import queryData from "../../src/repositories/database/queryData";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from "enzyme-to-json";
import axios from "../../src/redux/apis/axios";
Enzyme.configure({ adapter: new Adapter() });
// enableHooks(jest);
// jest.mock("react-redux", () => ({
//   ...jest.requireActual("react-redux"),
//   useSelector: jest.fn(),
//   useDispatch: jest.fn(),
// })); jest.mock("@react-native-google-signin/google-signin", () => ({
//   default: jest.fn(),
//   GoogleSignin: {
//     Configure: () => ({}),
//   }
// }));
// jest.mock("@react-navigation/native", () => ({
//   ...jest.requireActual("@react-navigation/native"),
//   useFocusEffect: jest.fn(),
//   useEffect: jest.fn(),
// }));
describe('queryData', () => {
  it("render component", () => {
    const wrapper = shallow(<queryData/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  })
  it("test Touchable Opacity", () => {
    const mockFn = jest.fn();
    const wrapper = shallow(<queryData function={mockFn} />);
   // wrapper.find("#touch").props().onPress();
    expect(mockFn).toHaveBeenCalled;
  })
describe("test querydata",()=>{
    it("test getusermeds",async ()=>{
        const payload="payload"
        jest.spyOn(axios,"get").mockImplementation(
            jest.fn(()=>Promise.resolve({data:"dfghjk"})))
            queryData.getusermeds(payload)
    })
})
});