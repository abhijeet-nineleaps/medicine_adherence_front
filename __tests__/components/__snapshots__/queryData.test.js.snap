// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queryData render component 1`] = `<queryData />`;
